src/Entity/Payment.php

```diff
@@ -2,192 +2,148 @@
 
 namespace App\Entity;
 
+use ApiPlatform\Doctrine\Orm\Filter\OrderFilter;
+use ApiPlatform\Doctrine\Orm\Filter\RangeFilter;
+use ApiPlatform\Doctrine\Orm\Filter\SearchFilter;
+use ApiPlatform\Metadata\ApiFilter;
 use ApiPlatform\Metadata\ApiResource;
 use ApiPlatform\Metadata\Delete;
 use ApiPlatform\Metadata\Get;
 use ApiPlatform\Metadata\GetCollection;
 use ApiPlatform\Metadata\Post;
 use ApiPlatform\Metadata\Put;
 use App\Repository\PaymentRepository;
-use Doctrine\Common\Collections\ArrayCollection;
-use Doctrine\Common\Collections\Collection;
+use App\Validator\Constraints\EventMember;
 use Doctrine\DBAL\Types\Types;
 use Doctrine\ORM\Mapping as ORM;
+use Symfony\Component\Serializer\Annotation\Groups;
+use Symfony\Component\Validator\Constraints as Assert;
 
 #[ApiResource(
     operations: [
         new Get(),
         new GetCollection(),
-        new Post(),
-        new Put(),
-        new Delete(),
-    ]
+        new Post(
+            denormalizationContext: ['groups' => 'payment:create'],
+        ),
+        new Put(
+            denormalizationContext: ['groups' => 'payment:update']
+        ),
+        new Delete()
+    ],
+    formats: ['json'],
+    denormalizationContext: [
+        'groups' => [
+            'payment:create',
+            'payment:update'
+        ]
+    ],
+    collectDenormalizationErrors: true
 )]
+#[ApiFilter(filterClass: SearchFilter::class, properties: ['event', 'member', 'name' => 'partial'])]
+#[ApiFilter(filterClass: RangeFilter::class, properties: ['date'])]
+#[ApiFilter(filterClass: OrderFilter::class, properties: ['date' => 'ASC', 'name' => 'ASC'])]
 #[ORM\Entity(repositoryClass: PaymentRepository::class)]
 class Payment
 {
-
     #[ORM\Id]
     #[ORM\Column(length: 255, unique: true)]
+    #[Assert\NotBlank]
+    #[Groups(['payment:create'])]
     private string $uid;
 
-    #[ORM\Column(length: 255)]
-    private string $name;
-
-    #[ORM\Column]
-    private int $amount;
-
-    #[ORM\Column(type: Types::DATETIME_MUTABLE)]
-    private ?\DateTimeInterface $date = null;
-
     #[ORM\ManyToOne]
-    #[ORM\JoinColumn(name: 'event_uid', referencedColumnName: 'uid', nullable: false)]
-    private PartyEvent $eventUid;
+    #[ORM\JoinColumn(name: 'event_uid', referencedColumnName: 'uid', nullable: false, onDelete: "CASCADE")]
+    #[Assert\NotBlank]
+    #[Groups(['payment:create'])]
+    private PartyEvent $event;
 
     #[ORM\ManyToOne]
-    #[ORM\JoinColumn(name: 'user_uid', referencedColumnName: 'uid', nullable: false)]
-    private User $userUid;
-
-    #[ORM\OneToMany(mappedBy: 'paymentUid', targetEntity: CalculationModification::class, orphanRemoval: true)]
-    private Collection $calculationModifications;
+    #[ORM\JoinColumn(name: 'member_uid', referencedColumnName: 'uid', nullable: false, onDelete: "CASCADE")]
```

Comment:
### ❗ Issue: So... if I delete some member, u will remove all payments? 

I don't think this is good idea... u can simply just set `'null` here